File: 02_selection_sort/golang/01_select_sort.go
package main

import (
	"fmt"
)

func  findSmall(arr []int) int {
	small := arr[0]
	small_index :=0
	for i:=0;i<len(arr);i++ {
		if arr[i] < small{
			small = arr[i]
			small_index =i
		}
	}
	return small_index
}

func selectSort(arr []int) []int{
	size := len(arr)
	newArr := make([]int,size)
	for i:=0;i<size ;i++ {
		small := findSmall(arr)
		newArr[i] = arr[small]
		arr = append(arr[:small],arr[small+1:]...)
	}
	return newArr
}

func main()  {
	s := []int{4,5,7,1,2,6,21}
	fmt.Println(selectSort(s))  //[1 2 4 5 6 7 21]
}

File: 02_selection_sort/php/01_select_sort.php
<?php
/**
 * Created by PhpStorm.
 * User: yanhao
 * Date: 2018/9/13
 * Time: 下午1:51
 */
//返回最小的数和下标
function fandSmall(array $arr){
    $small = $arr[0];
    $smallIndex = 0;
    for ($i=0;$i<count($arr);$i++){
        if ($arr[$i]<$small){
            $small = $arr[$i];
            $smallIndex= $i;
        }
    }
    return $smallIndex;
}

function selectionSort(array $arr){
    $newArr = [];
    for($i=0,$lenght= count($arr);$i<$lenght;$i++){
        $small = fandSmall($arr);
        array_push($newArr,array_splice($arr,$small,1)[0]);
    }
    return $newArr;
}


var_dump(selectionSort([5,2,1,6,7,8,9,10,22,42]));